scripts/prettier/index.js

```diff
@@ -28,23 +29,11 @@ const defaultOptions = {
 };
 const config = {
   default: {
-    patterns: [
-      // Internal forwarding modules
-      'packages/*/*.js',
-      // Source files
-      'packages/*/src/**/*.js',
-      'packages/shared/**/*.js',
-    ],
+    patterns: es6Path,
     ignore: ['**/node_modules/**'],
```

Comment:
Should ignore patterns also be a part of the respective configs? That would make sense to me. The fact that they’re currently duplicated in eslintignore is a bit confusing.

Reply:
OK, I'll move it to respective config.
Does it mean we should put all the `eslintignore` setting to the shared config file?

Reply:
I guess that could mess with the editor integrations people on the team might be using. So maybe we shouldn’t do it. What do you think?

Reply:
I think we should keep the `.eslintrc` to make editor work as usual. 
If we want to avoid ignore pattern duplicate, maybe we can write a script to read `.eslintrc` to use in `prettier` ignore part.

Reply:
That sounds a bit convoluted.

How about:

* We use top level eslintignore/prettierignore for things that should never be touched (like build products and third party code)
* For ES5/ES6, we always negate the other pattern to get a list of extra ignores. For example ES5 = everything in ES5 patterns except anything that is ES6. And vice versa.

Reply:
It’s even simpler.

ES6 doesn’t need a pattern. It should just be “everything”. The global ignores (in ESLint and Prettier top lebel ignore files) should filter out build artefacts and dependencies.

ES5 needs a pattern that just says “these files are ES5”. It’s a subset of all files.

You run ES5 passing it as “include” set.

Then you run ES6 by passing ES5 as “exclude” set. Because everything else is ES6 (or already excluded by global config). 

In either case ES5 is the only pattern that needs to be explicitly specified.

Reply:
Cool! It's more clear. I will correct it in this way.

BTW, for the lint change, eslint provide a `ignorePattern` params.
So when linting es5 files, we can pass es6 path as ignore pattern, vice versa.

However, it will cause this warning when the file is ignored: (related issue: [https://github.com/eslint/eslint/issues/5623](https://github.com/eslint/eslint/issues/5623))
```
0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
```
To turn off this warning is to set `quiet` as true, but it will quiet "all warning". I think it's inappropriate.
Therefore, I think write the filter to distinguish es5 files and es6 files is needed. I'll try a cleaner method to do it.

Reply:
I think we can just filter this particular warning from the output (and maybe raise an issue to ask for an option to completely disable this warning).